src/webview/components/memory-table.tsx

```diff
@@ -226,7 +237,49 @@ export class MemoryTable extends React.PureComponent<MemoryTableProps, MemoryTab
     }
 
     protected onSelectionChanged = (event: DataTableSelectionCellChangeEvent<MemoryRowData[]>) => {
-        this.setState(prev => ({ ...prev, selection: event.value }));
+        // eslint-disable-next-line no-null/no-null
+        const value = event.value ? event.value as MemoryTableCellSelection : null;
+        if (value) {
+            value.textContent = event.originalEvent.currentTarget?.textContent ?? '';
+        }
+
+        this.setState(prev => ({ ...prev, selection: value }));
+    };
+
+    protected onCopy = (_event: React.ClipboardEvent) => {
+        // if we don't have a text selection, we copy the textContent of the currently selected cell
+        if (!window.getSelection()?.toString() && this.state.selection) {
+            navigator.clipboard.writeText(this.state.selection.textContent);
+        }
+    };
+
+    protected onContextMenu = (event: React.MouseEvent) => {
+        /* Context menu events for cells are triggered on the root td.
+         * The rendered child content might provide additional vscode context data.
+         * Per default this child context will not be available in the merged vscode context
+         * So we need to handle this case manually and merge the child context into the parent context.
+         */
+        if (!(event.target instanceof HTMLElement)) {
+            return;
+        }
+        const parent = event.target.closest('.p-selectable-cell');
+        if (!parent || !(parent instanceof HTMLTableCellElement)) {
+            return;
+        }
+        let currentElement: Element | undefined = event.target;
+        const childContexts: object[] = [];
+        while (currentElement && currentElement !== parent) {
+            if (currentElement instanceof HTMLElement && currentElement.dataset?.vscodeContext) {
+                childContexts.push(JSON.parse(currentElement.dataset['vscodeContext']));
+            }
+            currentElement = currentElement.parentElement ?? undefined;
+        }
+
+        if (childContexts.length > 0) {
+            const parentContext = JSON.parse(parent.dataset['vscodeContext'] ?? '{}');
+            const mergedContext = childContexts.reduce((acc, cur) => ({ ...acc, ...cur }), parentContext);
+            parent.dataset['vscodeContext'] = JSON.stringify(mergedContext);
+        }
```

Comment:
Is there any way this could leak incorrect data if context menu events were triggered on two children of the same parent?